src/full_energy.c

```diff
@@ -206,26 +237,179 @@ int  full_energy(int                  gridcell,
   atmos->out_rain = 0;
   atmos->out_snow = 0;
 
-  /* Assign current veg albedo and LAI */
+  // Crop fraction
+  if (options.CROPFRAC) { //true or false set in global file, default=false
+
+    if (rec >= 0) {
+      for(iveg = 0; iveg < Nveg; iveg++){
+	if (veg_con[iveg].crop_frac_active) {
+	  veg_class = veg_con[iveg].veg_class;
+	  for ( band = 0; band < Nbands; band++ ) {
+	    if (rec == 0)
+	      old_crop_frac = veg_hist[rec][iveg].crop_frac[0];
+	    else
+	      old_crop_frac = veg_var[iveg][band].crop_frac;
+	    if(snow[iveg][band].swq<0.0001 && atmos->air_temp[NR]>7)  //ingjerd added. ok for all bands, veg types etc???? should be same air temp as below! (7 degrees)
+	      new_crop_frac = veg_hist[rec][iveg].crop_frac[0];
+	    else new_crop_frac = old_crop_frac;
+	    moistlayer=moisttotal1=moisttotal2=0.;
+	    if (new_crop_frac != old_crop_frac) {
+	      // The portion that grows needs to assimilate state variables
+	      // from the portion that shrinks; canopy storages need to be rescaled
+	      if (new_crop_frac > old_crop_frac) { // crop fraction is growing
+		for(lidx=0;lidx<options.Nlayer;lidx++) {
+		  moistlayer=all_vars_crop->cell[0][band].layer[lidx].moist*(1-old_crop_frac)+all_vars_crop->cell[1][band].layer[lidx].moist*old_crop_frac;
+		  moisttotal1+=moistlayer;
+		  all_vars_crop->cell[1][band].layer[lidx].moist = (all_vars_crop->cell[1][band].layer[lidx].moist*old_crop_frac+all_vars_crop->cell[0][band].layer[lidx].moist*(new_crop_frac-old_crop_frac))/new_crop_frac;
+		  all_vars_crop->cell[0][band].layer[lidx].moist = all_vars_crop->cell[0][band].layer[lidx].moist;
```

Comment:
This line seems redundant - is it just to account for the fact that cell[0] doesn't change?  Perhaps this could be replaced by a comment that cell[0] isn't changing...